### PR TITLE
CocoaPods support

### LINKED ISSUE #48: CocoaPods Support
Hey guys! super cool initiative and amazing work!

I wanted to ask if you guys plan to add support for CocoaPods any time soon?

Thanks!

### PR DESCRIPTION
**Pull request by ohitsdaniel**
Adding support for CocoaPods. This required to add three separate podspecs, one for the C++ helpers, one for OpenCombine and one for OpenCombineDispatch. Created a local example project and tested that it build with both dynamic frameworks and static libraries. ✅ 

As `os_unfair_lock` is only available on versions above the minimum deployment target, warnings are produced. There is a check in place that they're actually not used when they're not available. Therefore passing `--allow-warnings` to `pod lib lint` should be fine.

This PR does not expose the tests via a test_spec as this would've required adding a podspec to `GottaGoFast`. As this dependency was recently dropped, let's tackle this in a future PR. 

Fixes #48

### CODE REVIEW

File: .circleci/config.yml
Comment (broadwaylamb) [inline .circleci/config.yml]: The CI has [failed](https://app.circleci.com/jobs/github/broadwaylamb/OpenCombine/379) on this step. Seems like CocoaPods is already installed there. Can you remove this step and see whether it fixes the failure?

File: .circleci/config.yml
Comment (broadwaylamb) [inline .circleci/config.yml]: We don't have any dependencies, why is this needed?

**Review by broadwaylamb — CHANGES_REQUESTED** [review]
Thank you for taking this!

File: .circleci/config.yml
Comment (ohitsdaniel) [reply .circleci/config.yml]: Noticed this as well. I migrated the travis.yml in the old PR which included this command but I agree, we don't have dependencies, we don't need it.

File: .circleci/config.yml
Comment (ohitsdaniel) [reply .circleci/config.yml]: https://circle-macos-docs.s3.amazonaws.com/image-manifest/v1532/index.html

Exactly, I didn't know that the CircleCI macOS image has CocoaPods 1.8.4 pre-installed. Removing this step should fix the failing CI. :)

**Comment by broadwaylamb** [comment]
It looks like everything's working! 🎉🎉🎉

I'm curious though, do we need 3 separate podspecs, can't we use subspecs instead?

**Comment by ohitsdaniel** [comment]
@broadwaylamb 
Addressed your comments and added an installation guide to the README. :)

File: .circleci/config.yml
Comment (broadwaylamb) [inline .circleci/config.yml:229]: I've opened #104 to fix the warnings. I'll merge it as soon as the CI is passed, so we can remove the `--allow-warnings` flag.

**Comment by ohitsdaniel** [comment]
We need 3 separate podspecs because SPM and Cocoapods treat (sub)targets differently. Subspecs are not separate targets, but basically scoped views into the same target. Subspecs depending on other subspecs just increase the scope.

Cocoapods: 
A spec contains all it's subspecs and each subspec is part of the same parent module. Therefore it does not generate a it's own module and does not have it's own namespace. Importing code using the module name `COpenCombineHelpers` for example doesn't works as Cocoapods sees treats subspecs as parts of the same module. 

SPM targets:
We have three different target which produce an output (i.e. a library). The targets have dependencies on each other and import the code using the target output. 

We can get the same behaviour as in SPM by using three podspecs. Each podspec produces a library which can be included by the other specs.

File: .circleci/config.yml
Comment (ohitsdaniel) [reply .circleci/config.yml:229]: I think we should keep `--allow-warnings`. Lint also creates warnings when the Description is short than the summary. This would then lead to a failing CI job. Nevertheless, doing some cleanup and silencing those warnings sounds great. 👍

File: README.md
Comment (broadwaylamb) [inline README.md]: Actually there are 3 targets, one of them is just not really "public". I think it's worth mentioning that:

```markdown
`OpenCombine` contains two public targets: `OpenCombine` and 
`OpenCombineDispatch` (the third one, `COpenCombineHelpers`,
is considered private, don't import it in your projects)
```

Also, it would be good to mention what `OpenCombineDispatch` is, as [this can be non-obvious](https://github.com/broadwaylamb/OpenCombine/issues/94#issuecomment-546593378). Like so:

```markdown
OpenCombine itself doesn't depend on anything but the platform libraries.
Not even Foundation or Dispatch. For using OpenCombine with Dispatch
(for example for using `DispatchQueue` as `Scheduler` for operators
like `debounce`, `receive(on:)` etc.) you need to import both `OpenCombine`
and `OpenCombineDispatch`.
```

The latter note should probably go to the "Usage" section, which needs to be created first.

File: README.md
Comment (broadwaylamb) [inline README.md:21]:
```suggestion
To add `OpenCombine` to your [SPM](https://swift.org/package-manager/) package, add the `OpenCombine` package to the list of packages and target dependencies in your `Package.swift` file.
```

I think we should also mention Xcode, as many users will probably want to use the SPM integration:

```markdown
You can add OpenCombine as an SPM dependency right in your Xcode project (requires Xcode 11).
For that use **File** → **Swift Packages** → **Add Package Dependency…**, enter the repository URL,
choose the latest available version, and don't forget to activate the checkboxes:
<img alt="Select the OpenCombine and OpenCombineDispatch targets" src="https://user-images.githubusercontent.com/16309982/67618468-bd379f80-f7f8-11e9-917f-e76e878a1aee.png">
```

File: README.md
Comment (broadwaylamb) [inline README.md]:
```suggestion
    .target(name: "MyAwesomePackage", dependencies: ["OpenCombine", "OpenCombineDispatch"])
```

🙂

File: README.md
Comment (ohitsdaniel) [reply README.md]: Good points, I'll add them. 👍

**Comment by ohitsdaniel** [comment]
@broadwaylamb addressed your comments and worked a bit on the wording as well. I hope that's okay for you. :)

File: README.md
Comment (broadwaylamb) [inline README.md]:
```suggestion
To do so, open Xcode, use **File** → **Swift Packages** → **Add Package Dependency…**, enter the repository URL (https://github.com/broadwaylamb/OpenCombine.git), choose the latest available version, and activate the checkboxes:
```

File: README.md
Comment (broadwaylamb) [inline README.md]:
```suggestion
OpenCombine itself does not have any dependencies. Not even Foundation or Dispatch. If you want to use OpenCombine with Dispatch (for example for using `DispatchQueue` as `Scheduler` for operators like `debounce`, `receive(on:)` etc.), you will need to import both `OpenCombine` and `OpenCombineDispatch`.
```

File: README.md
Comment (broadwaylamb) [inline README.md]:
```suggestion
To add `OpenCombine` to a project using [CocoaPods](https://cocoapods.org/), add `OpenCombine` and `OpenCombineDispatch` to the list of target dependencies in your `Podfile`. 
```

**Comment by ohitsdaniel** [comment]
@broadwaylamb updated ✔️

**Comment by ohitsdaniel** [comment]
@broadwaylamb any other changes needed?

File: README.md
Comment (broadwaylamb) [inline README.md:47]: If we're merging this now, these lines will be shown in the readme, but they won't work, as the 0.5.0 release was not uploaded to the CocoaPods trunk.

Can you please change them to these instead so that users can start using CocoaPods immediately and don't have to wait for a tagged release?

```ruby
pod 'OpenCombine', :git => 'https://github.com/broadwaylamb/OpenCombine.git', :branch => 'master'
pod 'OpenCombineDispatch', :git => 'https://github.com/broadwaylamb/OpenCombine.git', :branch => 'master'
```

**Review by broadwaylamb — COMMENTED** [review]
Sorry for the delay. I wanted to test this on my local machine which I haven't had access to in the last couple of days

Can you please rebase onto the newest master and address the comment? After that I'm ready to merge!

File: README.md
Comment (ohitsdaniel) [reply README.md:47]: https://cocoapods.org/pods/OpenCombine
https://cocoapods.org/pods/OpenCombineDispatch
https://cocoapods.org/pods/COpenCombineHelpers

I pushed them to the CocoaPods trunk already and added you as contributor. Also tested this with a local example project and everything was working fine. :)

Comment (ohitsdaniel) [reply README.md:47]: Also, if this wouldn't work, `pod lib lint` would fail.

Comment (broadwaylamb) [reply README.md:47]: AFAIK `pod lib lint` just verifies the local setup, it never goes to the trunk.

Comment (broadwaylamb) [reply README.md:47]: >I pushed them to the CocoaPods trunk

I wish you asked me first :(

Comment (ohitsdaniel) [reply README.md:47]: Sorry about that! Had to test the setup and therefore pushed to trunk. Should've asked before that! If you want, I can delete them from the trunk and you can push them yourself. :)

**Comment by ohitsdaniel** [comment]
@broadwaylamb do you want me to remove the spec from the trunk so that you can push it yourself? Sorry for not asking you in advance. :/

**Comment by broadwaylamb** [comment]
@ohitsdaniel nevermind, let's leave it as-is.

**Pull request merged**
